virttest/qemu_devices/qcontainer.py

```diff
@@ -1720,6 +1720,147 @@ def images_define_by_params(self, name, image_params, media=None,
                                                    "bus_extra_params"),
                                                image_params.get("force_drive_format"))
 
+    def serials_define_by_variables(self, serial_id, serial_type, backend,
+                                    port_type=None, file_name=None, serial_name=None,
+                                    bus=None, nr=None, reg=None, **kwargs):
+        """
+        Creates related devices by variables
+
+        :param serial_id: the id of the serial device
+        :param serial_type: the type of the serial device
+        :param backend: the backend type of chardev device
+        :param port_type: port type of the serial device(optional, virtio only)
+        :param path: the path option of chardev device(optional)
+        :param name: the name option of serial device(optional)
+        :param bus: the busid of parent bus(optional, virtio only)
+        :param nr: the nr(port) of the parent bus(optional, virtio only)
+        :param reg: reg option of isa-serial(optional)
+        :keyword chardev options per backend
+        :return: the device list that construct the serial device
+        """
+        if (not self.has_option('chardev') or
+                not self.has_device(serial_type)):
```

Comment:
@yanan-fu I think it should get `True` for `not self.has_device(serial_type)`, for the default definition `serial_type = "isa-serial"`, so it get `-serial` by L1743

Reply:
@vivianQizhu  `I think it should get True for not self.has_device(serial_type)`  ?   do you mean are do not have `isa-serial` device ?

The old logic is,  arm have `isa_serial` but it can not work normally, so use `- serial` instead.

Reply:
@yanan-fu Really? Thanks for the information, so is that a workaround for the product issue? I will modify the code to handle it if so.

Reply:
@vivianQizhu  please refer the old comment, and if we do not test with other arch,  we should keep old logic.

Reply:
@yanan-fu You are right, I will update it. Thanks.